tests/test_state.cc

```diff
@@ -127,304 +128,212 @@ void doTest(core::State& s) {
   doSimpleTest(s);
 }
 
-// TODO: there should be better way (using gtest?) than writing a main
-// this is just for demo purpose
-// After compilation, run test_state from build folder to run the test
-int main() {
-  int seed = 999;
 
-  {
-    std::cout << "testing: tristannogo" << std::endl;
+constexpr int seed = 999;
```

Comment:
is the purpose of using a constexpr here to simply make it a teeeeeeny bit faster?

Reply:
It's a compile-time constant that doesn't change. I didn't really think about it :)